lib/tasks/trn_generator.rb

```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+class TRNGenerator
+  class << self
+    def next
+      available.pop || (raise "TRN available list exhausted")
+    end
+
+  private
+
+    ALL_TRNS = ("0011111".."9999999").to_a.freeze unless defined?(ALL_TRNS)
+
+    def available
+      @available ||= (ALL_TRNS - taken).shuffle
+    end
```

Comment:
I wonder how much we need the shuffle. What's the risk of generating TRNs sequentially? (which would presumably allow us not to keep track of an array of strings couple of millions long)

Reply:
We only ever look at the head of the array, so internally it's a single pointer update. Not hard to keep track of what essentially is a stack.